## Урок 1. ДЗ (17.02.2021)
## Теория вероятности
### Случайные события. Условная вероятность. Формула Байеса.  Независимые испытания

* [Теория](#theor)
* [Практическое задание. ДЗ.](#ex)
    * [Задание 1](#ex1)
    * [Задание 2](#ex2)
    * [Задание 3](#ex3)
    * [Задание 4](#ex4)

# <center>Теория <a class="anchor" id="theor"></a><center>

**Случайное событие** — то, которое при определенных условиях может произойти или нет.



Примеры случайного события:
1. При броске двух игральных костей на одной выпало число 1, а на другой — 2.
2. Клиент банка не вернул кредит.
3. Температура воздуха в Москве за последние десять дней не превышала 29 градусов по Цельсию.
4. Монету подбросили сто раз, орел выпал в 55 случаях. 

Событие можно называть **достоверным**, если в результате испытания оно обязательно произойдет.

**Невозможное** событие не произойдет никогда.

Примеры достоверного события:
1. На игральной кости выпало число, не превышающее 6.
2. Подбросили монету, и выпал либо орел, либо решка.
3. Монету подбросили стократно, и решка выпала не более 100 раз.

Примеры невозможного события:
1. Две игральные кости бросили один раз, и сумма выпавших чисел составила 15.
2. Монету подбросили стократно, и решка выпала 55 раз, а орел — 56.
3. Три игральные кости бросили один раз, и сумма выпавших чисел составила 2.

Для случайного события есть понятие **относительной частоты** — это отношение количества состоявшихся событий к общему числу испытаний:

$$W(A) = \frac{m}{n}$$

В формуле $W(A)$ — это относительная частота события $A$;

$m$ — число состоявшихся событий $A$;

$n$ — общее число испытаний.

Рассмотрим примеры случайных событий.

**Пример 1**

Смоделируем 60-кратное подбрасывание игральной кости с помощью функции [random.randint](https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.random.randint.html) пакета **numpy**, то есть $n$ = 60.

Событием $A$ будем считать выпадение числа 3 и найдем его относительную частоту.

In [ ]:
import numpy as np

n = 60

b = np.random.randint(1, 7, size=n)
b

array([2, 2, 4, 6, 3, 5, 5, 5, 4, 2, 1, 5, 2, 5, 5, 1, 4, 2, 5, 1, 5, 1,
       1, 2, 5, 2, 5, 6, 6, 4, 4, 5, 3, 5, 2, 2, 3, 6, 3, 6, 1, 4, 1, 2,
       5, 1, 2, 1, 3, 2, 4, 3, 1, 6, 1, 3, 2, 2, 2, 4])

Вычислим мощность подмножества, где в результате испытания выпадало число 3, то есть наблюдалось событие $A$:

In [ ]:
m = len(b[b==3])
m

7

Теперь можем вычислить относительную частоту события $A$:

In [ ]:
W = m / n
W

0.11666666666666667

**Пример 2**

Разберем более сложный пример. Смоделируем ситуацию, когда бросают две игральные кости.

При этом будем находить частоту случайного события $B$, при котором на первой кости выпало 1, а на второй — 2. 

Проведем для этого 360 испытаний. Сразу зададим число $n$:

In [ ]:
n = 360

In [ ]:
c = np.random.randint(1, 7, size=n)
d = np.random.randint(1, 7, size=n)

In [ ]:
c

array([6, 1, 1, 3, 2, 2, 6, 5, 3, 1, 1, 1, 3, 2, 4, 3, 1, 2, 5, 5, 3, 2,
       6, 6, 1, 4, 1, 3, 3, 2, 1, 3, 2, 3, 2, 4, 5, 3, 3, 2, 5, 4, 5, 4,
       2, 6, 6, 5, 3, 6, 5, 1, 4, 6, 4, 4, 2, 3, 4, 1, 3, 1, 6, 3, 1, 6,
       3, 4, 3, 3, 6, 4, 4, 3, 6, 4, 3, 2, 1, 2, 1, 5, 1, 4, 4, 5, 1, 5,
       2, 4, 2, 4, 3, 5, 3, 6, 1, 6, 1, 1, 2, 1, 2, 1, 5, 6, 3, 2, 3, 5,
       3, 5, 5, 4, 3, 4, 1, 3, 2, 1, 3, 2, 4, 4, 4, 4, 1, 1, 6, 3, 6, 2,
       4, 6, 2, 4, 5, 4, 3, 1, 6, 3, 2, 2, 3, 1, 2, 6, 4, 1, 1, 1, 2, 2,
       6, 3, 3, 3, 4, 1, 6, 2, 2, 1, 2, 3, 5, 1, 4, 1, 2, 5, 5, 4, 1, 3,
       5, 4, 5, 2, 5, 3, 2, 1, 5, 1, 3, 6, 5, 3, 4, 3, 3, 3, 6, 5, 6, 6,
       5, 2, 1, 3, 2, 4, 4, 4, 3, 1, 2, 1, 5, 3, 2, 6, 3, 3, 2, 5, 3, 1,
       2, 5, 2, 3, 3, 3, 3, 4, 6, 3, 6, 4, 6, 2, 1, 6, 6, 6, 1, 3, 1, 4,
       6, 6, 4, 4, 5, 5, 5, 1, 6, 5, 2, 4, 2, 5, 5, 5, 3, 1, 5, 6, 6, 3,
       3, 5, 5, 3, 4, 4, 1, 6, 3, 6, 3, 2, 4, 2, 5, 2, 3, 1, 1, 6, 6, 4,
       2, 3, 4, 4, 4, 5, 6, 5, 1, 1, 2, 1, 6, 4, 3,

In [ ]:
d

array([6, 2, 4, 5, 5, 4, 4, 1, 1, 4, 6, 3, 3, 2, 1, 4, 6, 2, 4, 1, 2, 4,
       6, 4, 5, 4, 4, 4, 4, 2, 4, 2, 2, 6, 2, 2, 3, 1, 5, 2, 1, 1, 2, 3,
       2, 1, 4, 1, 1, 5, 2, 5, 5, 1, 4, 6, 6, 2, 2, 5, 6, 2, 1, 3, 6, 1,
       2, 4, 1, 2, 5, 2, 5, 6, 1, 4, 5, 3, 2, 4, 3, 6, 3, 6, 1, 3, 2, 5,
       1, 6, 6, 2, 1, 5, 2, 1, 5, 1, 4, 2, 5, 6, 5, 6, 6, 1, 5, 1, 1, 2,
       6, 1, 3, 4, 3, 1, 5, 4, 3, 5, 6, 4, 1, 2, 6, 5, 2, 4, 6, 1, 3, 4,
       5, 5, 6, 6, 5, 5, 2, 1, 2, 1, 4, 1, 3, 6, 2, 6, 1, 3, 4, 5, 4, 5,
       3, 3, 4, 4, 4, 3, 2, 2, 1, 3, 3, 2, 2, 2, 4, 3, 3, 4, 5, 4, 2, 2,
       6, 6, 6, 2, 3, 3, 1, 6, 2, 3, 3, 6, 2, 2, 6, 5, 5, 1, 5, 2, 2, 5,
       4, 5, 3, 6, 4, 5, 6, 5, 6, 4, 2, 6, 2, 6, 4, 5, 1, 3, 1, 1, 5, 3,
       1, 6, 1, 6, 3, 5, 3, 1, 1, 4, 2, 5, 5, 1, 4, 6, 5, 5, 6, 5, 1, 4,
       5, 5, 2, 3, 5, 6, 2, 3, 4, 4, 4, 1, 6, 5, 1, 5, 3, 4, 5, 6, 2, 5,
       4, 4, 5, 4, 5, 5, 2, 4, 5, 3, 5, 4, 4, 5, 1, 6, 3, 2, 3, 5, 1, 1,
       4, 6, 4, 6, 2, 4, 4, 4, 5, 2, 2, 5, 1, 2, 3,

Числа, стоящие на одной позиции в массивах $c$ и $d$, будем считать результатами, полученными в одном испытании.

В первом испытании на игральной кости № 1 выпало число $c[0]$, а на № 2  — $d[0]$.

In [ ]:
c[0]

6

In [ ]:
d[0]

6

Посчитаем число случав, когда в одном испытании на первой игральной кости выпало число 1, а на второй — 2.

In [ ]:
a = c[(c==1) & (d==2)]
m = len(a)
m

11

Вычислим относительную частоту события $B$:

In [ ]:
W = m / n
W

0.030555555555555555

#### Статистическая вероятность

При достаточно большом количестве испытаний $n$ величина относительной частоты $W$ будет стремиться к конкретному числу. Оно называется **статистической вероятностью** и обозначается как $P(A)$:

$$P(A) = \frac{m}{n}$$

Статистическую вероятность можно вычислить на основе данных, полученных по результатам многочисленных испытаний.

#### Классическое определение вероятности

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.activity.readonly&response_type=code

Enter your authorization code:



ValueError: ignored

Если заранее известны все вероятные исходы и они равновозможны (например, при подбрасывании монеты или игральной кости), можно воспользоваться **классическим определением вероятности**:

Вероятность события — это отношение числа элементарных исходов, благоприятствующих данному событию, к числу всех равновозможных исходов опыта, в котором оно может появиться.


Формула для этого определения такая же, как и для статистической вероятности:

$$P(A) = \frac{m}{n}$$

**Пример 3**

Вычислим вероятность того, что на игральной кости выпадет число 3. Известно, что у кости шесть граней с числами от 1 до 6 и каждая может выпасть с равной вероятностью. По формуле классической вероятности получаем результат:

$$P(A) = \frac{m}{n} = \frac{1}{6}$$

**Пример 4**

Вычислим, с какой вероятностью на игральной кости выпадет 2 или 4. Эти события в равной степени вероятны. Как следует из предыдущего примера, каждая из этих вероятностей равна 1/6. И эти события — несовместные, то есть не могут произойти одновременно. Поэтому мы можем складывать их вероятности:

$$P(A∨B) = \frac{1}{6} + \frac{1}{6} = \frac{2}{6}$$

($∨$ — знак логического ИЛИ).

Для более сложных случаев число исходов $k$, благоприятствующих данному событию, или число всех элементарных исходов испытания ($n$) можно вычислить с помощью формул комбинаторики.

#### Формулы комбинаторики

Число **сочетаний** из $n$ элементов по $k$ элементов в каждом (в сочетаниях порядок не важен):

$$C_n^k = \frac{n!}{k!(n - k)!}$$

Напишем функцию, чтобы вычислить количество сочетаний:

In [ ]:
from math import factorial

In [ ]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

Определим число **размещений** из $n$ элементов по $k$ элементов в каждом. При размещениях порядок важен, поэтому вариантов размещения может быть больше, чем сочетаний при заданных $k$  и $n$.

$$A_n^k = \frac{n!}{(n - k)!}$$

In [ ]:
def arrangements(n, k):
    return int(factorial(n) / factorial(n - k))

Число **перестановок** из $n$ элементов — при перестановках важен порядок, но отличие от размещений в том, что применяются все имеющиеся $n$ элементов:

$$P_n = n!$$

In [ ]:
def permutations(n):
    return int(factorial(n))

Рассмотрим на примерах, как применяются формулы комбинаторики.

**Пример 5**

Сколькими способами можно выбрать из колоды, состоящей из 36 карт, 4 карты?

Для ответа на этот вопрос воспользуемся формулой для вычисления количества сочетаний:

$$C_{36}^4 = \frac{36!}{4!(36 - 4)!} = \frac{36!}{4! \cdot 32!} = \frac{33 \cdot 34 \cdot 35 \cdot 36}{4!} = \frac{33 \cdot 34 \cdot 35 \cdot 36}{2 \cdot 3 \cdot 4} = 58905$$

или

In [ ]:
combinations(36, 4)

58905

**Пример 6**

В магазине 20 покупателей. Сколькими способами они могут образовать очередь из 5 человек?

В этом примере важен порядок, в котором покупатели будут стоять в очереди, поэтому применим формулу для нахождения числа размещений:

In [ ]:
arrangements(20, 5)

1860480

**Пример 7**

Сколькими способами 5 покупателей могут образовать очередь?

Этот пример похож на предыдущий, но есть важное отличие: не нужно выбирать 5 покупателей из 20. Здесь их всего 5, и все должны присутствовать в очереди. Применим формулу для нахождения числа перестановок:

In [ ]:
permutations(5)

**Пример 8**

Из колоды, состоящей из 36 карт, случайным образом выбраны 5. Сколькими способами можно выбрать эти карты так, чтобы среди них оказалось от 2 до 3 тузов?

Чтобы решить эту задачу, нужно сначала рассмотреть ситуацию, когда выбирается два туза из четырех.

Число таких сочетаний будет равно:

In [ ]:
combinations(4, 2)

NameError: ignored

Остальные три карты выбираются из 32 карт  — тузы не рассматриваются, так как они уже выбраны:

In [ ]:
combinations(32, 3)

И в этом случае будет такое число сочетаний, когда из колоды, в которой 36 карт, выбраны пять, две из которых — тузы:

In [ ]:
6 * 4960

Теперь рассмотрим ситуацию выбора трех тузов из четырех. Число сочетаний будет равно:

In [ ]:
combinations(4, 3)

Остальные две карты выбираются из 32 карт. Число сочетаний будет равно:

In [ ]:
combinations(32, 2)

И тогда будет такое число сочетаний, когда из колоды в 36 карт выбраны пять, три из которых — тузы:

In [ ]:
4 * 496

Осталось сложить полученные числа сочетаний:

In [ ]:
29760 + 1984

Число комбинаций, полученных с помощью формул комбинаторики, можно складывать 
и умножать. Сложение в приведенном примере соответствует логическому ИЛИ, а умножение — логическому И.

**Совместные и несовместные события**

Совместные события могут произойти в одном испытании, а несовместные — нет. Например, при однократном подбрасывании игральной кости не могут сразу выпасть числа 3 и 4 — эти события несовместны. Если бросают две кости, на одной может выпасть 4, а на другой — 5, это совместные события. Вероятности несовместных событий можно складывать:

$$P(A + B) = P(A) + P(B)$$

В случае совместных событий формула суммы событий другая: из суммы вероятностей отдельных событий вычитается вероятность их совместного появления.

$$P(A + B) = P(A) + P(B) - P(AB)$$

**Вероятность зависимых и независимых событий**

*Независимыми* события  называют, когда появление одного из них не влияет на появление другого. С зависимыми все наоборот.

Вероятность одновременного появления двух независимых событий вычисляется по формуле:

$$P(A*B) = P(A) * P(B)$$

Вероятность появления двух зависимых событий:

$$P(A*B) = P(A) * P(B\:|\:A) = P(B) * P(A\:|\:B)$$

**Пример 9**

Найдем вероятность того, что при броске двух игральных костей на первой выпадет четное число, а на второй — кратное трем. Сначала рассмотрим вероятность выпадения четного числа на первой кости. Таких случаев может быть три: это события, при которых выпадают числа 2, 4 и 6. Вероятность для каждого из них равна 1/6.

Так как эти три события являются несовместными, вероятность их суммы равна сумме их вероятностей:

$$P(четн.) = \frac{1}{6} + \frac{1}{6} + \frac{1}{6} = \frac{3}{6} = \frac{1}{2}$$

Для второй кости подходящих событий два: это выпадение чисел 3 и 6. Это тоже несовместные события, и мы можем сложить их вероятности:

$$P(кратн. 3) = \frac{1}{6} + \frac{1}{6} = \frac{2}{6} = \frac{1}{3}$$

С учетом того, что выпадение числа на одной кости не влияет на результат на другой — то есть эти события независимы, — применим умножение вероятностей и получим итоговый ответ:

$$P = \frac{1}{2} * \frac{1}{3} = \frac{1}{6}$$

**Формула полной вероятности**

Если событие $A$ может наступить только при появлении событий $B_1, B_2, \dots, B_n$, образующих полную группу несовместных событий*, то вероятность $A$ вычисляется по формуле:

***полная группа событий** означает, что при отдельном испытании обязательно возникнет одно из них.

$$P(A) = P(B_1) \cdot P(A\:|\:B_1) + P(B_2) \cdot P(A\:|\:B_2) \; + ... + \; P(B_n) \cdot P(A\:|\:B_n)$$

Выражение $P(B\;|\;A)$ означает вероятность наступления события $B$ при условии, что событие $A$ произошло.

**Пример 10**

Есть три одинаковые корзины. В первой лежат три красных и пять зеленых мячей, во второй — только красные, а в третьей — только зеленые мячи. Случайно выбирается одна корзина и из нее случайным образом извлекается мяч. 

Какова вероятность того, что этот мяч — зеленый?

Решим эту задачу с помощью формулы полной вероятности. Событием $A$ будем считать то, что достанут зеленый мяч, а событием $B$ — что выберут определенную корзину. Причем вероятность выбора одной из трех корзин будет равна 1/3.

Вероятность достать зеленый мяч при условии, что выбрана первая корзина, равен 5/(3 + 5), то есть 5/8. При выборе второй корзины вероятность вынуть зеленый мяч равна 0. В случае с третьей корзиной вероятность равна 1, так как все мячи в ней — зеленые.

Подставим данные значения в формулу полной вероятности и получим итоговый ответ:

$$P(A) = \frac{1}{3} \cdot \frac{5}{8} + \frac{1}{3} \cdot 0 + \frac{1}{3} \cdot 1 = \frac{5}{24} + 0 + \frac{1}{3} = \frac{13}{24}$$

#### Формула Байеса

Чтобы определить вероятность события $B$ при условии, что событие $A$ уже произошло, используют формулу Байеса:

$$P(B\:|\:A) = \frac{P(B) \cdot P(A\:|\:B)}{P(A)}$$

В данной формуле вероятность $P(B\;|\;A)$ называется апостериорной, то есть определенной после того, как произошло событие $A$. 

А вероятность $P(B)$ — априорной, определенной до испытания.

**Пример 11**

Соревнования по биатлону. Один из трех спортсменов стреляет и попадает в мишень. 

Вероятность такого события для первого спортсмена равна 0.2, для второго — 0.4, для третьего — 0.7.

Задача: найти вероятность того, что выстрелы произведены третьим спортсменом.

Прежде чем узнали результат выстрела, мы располагали только априорной вероятностью того, что выстрел был произведен, и она равнялась 1/3 — если учесть, что у всех спортсменов были равные возможности выстрелить.

Зная результат выстрела, мы можем заново оценить вероятности для каждого спортсмена — ведь точность у них разная. Эти вероятности будут апостериорными, так как они получены после поступления информации о том, что выстрел был удачным.

Событием $A$ будем считать попадание в мишень, а событиями $B_1, B_2$ и $B_3$ — что выстрел совершил первый, второй или третий спортсмен:

$$P(A\:|\:B_1) = 0.2$$

$$P(A\:|\:B_2) = 0.4$$

$$P(A\:|\:B_3) = 0.7$$

Вероятность того, что стрелял третий спортсмен — при условии, что выстрел был удачным, — будем находить по формуле:

$$P(B_3\:|\:A) = \frac{P(B_3) \cdot P(A\:|\:B_3)}{P(A)}$$

---



Все величины подставим из условия, а знаменатель дроби распишем по формуле полной вероятности:

$$P(B_3\:|\:A) = \frac{\frac{1}{3} \cdot 0.7}{\frac{1}{3} \cdot 0.2 + \frac{1}{3} \cdot 0.4 + \frac{1}{3} \cdot 0.7} = \frac{7}{13}$$

# <center>Практическое задание <a class="anchor" id="ex"></a><center>

### <center>Задание 1 <a class="anchor" id="ex1"></a><center>
Из колоды в 52 карты извлекаются случайным образом 4 карты.

a) Найти вероятность того, что все карты – крести.

б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

### Решение:

Находим количество комбинаций или общее число исходов:

In [70]:
from math import factorial

def combinations(n, k):
  return int(factorial(n) / (factorial(k) * factorial(n-k)))

print('Общее число исходов: ', combinations(52,4))

Общее число исходов:  270725


#### Общее число исходов:
### $$ C_{52}^{4} =  \frac{n!}{k!(n-k)!} = \frac{52!}{4!(52-4)!} = \frac{52!}{4!(48)!} = \frac{49 \cdot 50 \cdot 51 \cdot 52}{1 \cdot 2 \cdot 3 \cdot 4} = 270725$$

#### a) P - число исходов благоприятствующих событию / общее число исходов.
#### Число исходов благоприятствующих событию - это количество извлечения комбинаций крестей 4 карты из 52/4 = 13:

### $$C^4_{13}  = \frac{13!}{4!(13-4)!} = \frac{13!}{4! \cdot 9!} = \frac{10 \cdot 11 \cdot 12 \cdot 13}{4!} = 715$$

#### $$ P = \frac{C_{4}^{13}}{C_{4}^{52}} = \frac{715}{270725} = 0,26\%$$

In [69]:
print(combinations(52, 4))
print(combinations(13, 4))
print('Вероятность того, что все карты крести =', combinations(13, 4)/combinations(52, 4))

270725
715
Вероятность того, что все карты крести = 0.0026410564225690276


#### б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

#### Благоприятные исходы это выпадение 1 туза, 2, 3 и 4.

### $$ 52-4 = 48 $$

#### - при выпадении 1 туза количество благоприятных исходов
### $$ \frac{4 \cdot  48!}{3!(48-3)!} = \frac{4 \cdot  48!}{3! \cdot 45!} = \frac{4 \cdot  46  \cdot 47  \cdot 48}{3!} = 4 \cdot 17296 = 69184$$

#### - при выпадении 2 тузов количество благоприятных исходов
### $$ \frac{6 \cdot 48!}{2!(48-2)!} = 6 \cdot  1128=6768$$

#### - при выпадении 3 тузов количество благоприятных исходов
### $$ \frac{4 \cdot 48!}{1!(48-1)!} = 4 \cdot 48=192$$

#### - при выпадении 4 тузов количество благоприятных исходов

### $$ = 1 $$

#### Общее количесвто благоприятных исходов: 69184+6768+192+1 = **76145**

#### Общее количество возможных исходов: $ C_{52}^{4} = $ **270725**

#### Вероятность того, что окажется один туз: 76145/270725 = **28,13%**







### Python:
#### А - среди выбранных карт хотя бы один туз
### А = А1 + А2 + А3 + А4

#### А1 - среди выбранных один туз
#### А2 - среди выбранных два туза
#### А3 - среди выбранных три туза
#### А4 - среди выбранных четыре туза

#### $C_{52}^{4}$ - сочетание 4 из 52
#### P1 = (сочетание 1 из 4 ) * (сочетание 3 из 48)
#### P2 = (сочетание 2 из 4 ) * (сочетание 2 из 48)
#### P3 = (сочетание 3 из 4 ) * (сочетание 1 из 48)
#### P4 = сочетание 4 из 4

In [67]:
P1 = (factorial(4) / (factorial(1)*factorial(4 - 1))) * (factorial(48) / (factorial(3)*factorial(48 - 3))) / (combinations(52, 4))
P2 = (factorial(4) / (factorial(2)*factorial(4 - 2))) * (factorial(48) / (factorial(2)*factorial(48 - 2))) / (combinations(52, 4))
P3 = (factorial(4) / (factorial(3)*factorial(4 - 3))) * (factorial(48) / (factorial(1)*factorial(48 - 1))) / (combinations(52, 4))
P4 = factorial(4) / (factorial(4)*factorial(4 - 4)) / (combinations(52, 4))
P = P1 + P2 + P3 + P4
print('Вероятность того, что из 4 карт хотя бы один туз =', P)

Вероятность того, что из 4 карт хотя бы один туз = 0.2812632745405855


### <center>Задание 2 <a class="anchor" id="ex2"></a><center>
На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

### Решение:

### Благоприятный исход $ = 1$
### Общее число комбинаций $$C^3_{10}  = \frac{n!}{k!(n-k)!} = \frac{10!}{3!(10-3)!} = \frac{10!}{3!7!} = \frac{720}{6}=120$$
### Вероятность открыть с первой попытки $$1/120 = 0,83%$$

In [66]:
P310 = 1 / (combinations(10, 3))
print('Вероятность того, что дверь открыта с первой попытки = ', P310)

Вероятность того, что дверь открыта с первой попытки =  0.008333333333333333


### <center>Задание 3 <a class="anchor" id="ex3"></a><center>
В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

### Решение:

### 9 окрашенных деталей могут быть извлечены $$\frac{9!}{3!(9-3)!} = \frac{9!}{3! \cdot 6!} = \frac{9 \cdot 8 \cdot 7}{3!} = 84$$ способами, которые являются благоприятными исходами при общем количестве $$\frac{15!}{3!(15-3)!} = \frac{15!}{3! \cdot 12!} = \frac{15 \cdot 14 \cdot 13}{3!} = 455$$ вариантов извлечения, таким образом вероятность того,  что все детали будут окрашены $$P=\frac{84}{455} = 18.46\%$$

In [63]:
print('Вероятность того, что все извлеченные детали окрашены = ', combinations(9, 3) / combinations(15, 3))

Вероятность того, что все извлеченные детали окрашены =  0.18461538461538463


### <center>Задание 4 <a class="anchor" id="ex4"></a><center>
В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

### Решение:

### Благоприятный исход = 1 при общем количестве $$\frac{100!}{2!(100-2)!} = \frac{100!}{2! \cdot 98!} = \frac{100 \cdot 99}{2} = 4950$$ вариантов извлечения, таким образом вероятность что два приобретенных билета будут выигрышными $$P=\frac{1}{4950} = 0.02 \%$$

In [64]:
print('Вероятность того, что 2 билета окажутся выигрышными = 1 / ', combinations(100, 2), ' и ', 1/combinations(100, 2))

Вероятность того, что 2 билета окажутся выигрышными = 1 /  4950  и  0.00020202020202020202
